bigchaindb/common/transaction.py

```diff
@@ -613,33 +612,33 @@ def __init__(self, operation, asset, fulfillments=None, conditions=None,
                 not asset and operation != Transaction.CREATE):
             raise TypeError('`asset` must be an Asset instance')
 
-        if conditions and not isinstance(conditions, list):
-            raise TypeError('`conditions` must be a list instance or None')
+        if outputs and not isinstance(outputs, list):
+            raise TypeError('`outputs` must be a list instance or None')
 
-        if fulfillments and not isinstance(fulfillments, list):
-            raise TypeError('`fulfillments` must be a list instance or None')
+        if inputs and not isinstance(inputs, list):
+            raise TypeError('`inputs` must be a list instance or None')
 
         if metadata is not None and not isinstance(metadata, dict):
             raise TypeError('`metadata` must be a dict or None')
 
         self.version = version if version is not None else self.VERSION
         self.operation = operation
         self.asset = asset if asset else Asset()
-        self.conditions = conditions if conditions else []
-        self.fulfillments = fulfillments if fulfillments else []
+        self.outputs = outputs if outputs else []
+        self.inputs = inputs if inputs else []
         self.metadata = metadata
 
         # validate asset
         # we know that each transaction relates to a single asset
-        # we can sum the amount of all the conditions
+        # we can sum the amount of all the outputs
         # for transactions other then CREATE we only have an id so there is
         # nothing we can validate
         if self.operation == self.CREATE:
-            amount = sum([condition.amount for condition in self.conditions])
+            amount = sum([condition.amount for condition in self.outputs])
             self.asset.validate_asset(amount=amount)
 
     @classmethod
-    def create(cls, owners_before, owners_after, metadata=None, asset=None):
+    def create(cls, creators, recipients, metadata=None, asset=None):
```

Comment:
In a CREATE transaction, if the "owners_before" list is just one public key, then the associated private key must be used to produce a signature and that signature must be in the transaction. That public key may be unrelated to the "creator" of the asset. It might be the public key of some agent that submits transactions on behalf of the creator, a "transaction builder & submitter agent."

If the "owners_before" list contains m public keys, then there must be m corresponding signatures. Again though, those might not be "creators" of the asset in question.

Idea: Instead of renaming "owners_before" to "creators" here, how about "tx_signers"? That is, the list of signers of the transaction in question.

As for "recipients," that's probably okay, so long as one correctly interprets what it means to "receive," i.e. one receives not ownership, but transfer permissions.